osu.Game/Online/Rooms/MultiplayerBeatmapTracker.cs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Linq;
+using osu.Framework.Bindables;
+using osu.Game.Beatmaps;
+
+namespace osu.Game.Online.Rooms
+{
+    public class MultiplayerBeatmapTracker : DownloadTrackingComposite<BeatmapSetInfo, BeatmapManager>
+    {
+        public readonly IBindable<PlaylistItem> SelectedItem = new Bindable<PlaylistItem>();
+
+        /// <summary>
+        /// The availability state of the currently selected playlist item.
+        /// </summary>
+        public IBindable<BeatmapAvailability> Availability => availability;
+
+        private readonly Bindable<BeatmapAvailability> availability = new Bindable<BeatmapAvailability>();
+
+        public MultiplayerBeatmapTracker()
+        {
+            State.BindValueChanged(_ => updateAvailability());
+            Progress.BindValueChanged(_ => updateAvailability());
+            updateAvailability();
```

Comment:
this is potentially unsafe if any transforms are added in value change callbacks of `availability`.

Reply:
the code at fault would be the one binding to `availability` that early, it shouldn't do so until `LoadComplete`.

Reply:
value change callbacks propagate on the same thread on which the original thread that triggered the change happened, so i'm not sure how the consumer's load state is relevant?

Reply:
I don't follow where the issue is, `State` and `Progress` are both changed on the update thread?